packages/expo-dev-launcher/ios/EXDevLauncherController.m

```diff
@@ -327,6 +327,8 @@ - (void)navigateToLauncher
   [self _applyUserInterfaceStyle:UIUserInterfaceStyleUnspecified];
 
   [self _removeInitModuleObserver];
+  // Reset app react host
+  [[self.delegate rctAppDelegate].rootViewFactory setReactHost:nil];
```

Comment:
the `[self.delegate rctAppDelegate]` is weak reference so it could be nil, we should handle this case